D2IQ-68510: dates and time formatting guidelines

mperrotti opened the pull request:
Closes D2IQ-68510

<!-- See Checklist for PR creators below. -->

## Testing
Read the guidelines about how to deal with dates and times in D2iQ apps. I recommend reviewing using GitHub's markdown view.

Please comment if:
- anything in the guidelines is unclear
- you can think of scenarios that are not covered by these guidelines
- you disagree with anything from these guidelines

## Trade-offs
This first draft does not have any guidelines to account for translation/internationalization. I think we need to come up with some guidelines about designing for a global audience before we attempt to document how to handle global date and time formatting.

## Dependencies

<!--
What needs to happen before this can be merged? e.g. PRs merged, other events
-->

## Screenshots

<!--
Would a visual be helpful for reviewers? e.g. "Before" and "After", visual changes a designer can check before merge
-->

## Checklist for PR creator

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [x] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [x] Info for applicable sections above is provided

GeorgiSTodorov commented on design-guidelines/copy-and-content/date-time-formatting.md:
```diff
@@ -0,0 +1,162 @@
+
+## Referring to dates and times
+
+When writing copy that refers to dates or times, use the same kind of language you would use in written communication.
```
What does "when writing copy" mean?

mperrotti replied:
When writing any text that appears in the app. I should find a better way to say this 👍

danielone replied:
"When writing copy for instruction text, descriptions and longer text, use either the approved abbreviation or full format as best for the appropriate context so the text reads clearly. 

[examples you have below]

"For tables, cards, and other components where space is at a premium, default should use the abbreviated format to maximize spacing, efficiency in readability, and consistency across our data tables."

GeorgiSTodorov commented on design-guidelines/copy-and-content/date-time-formatting.md:
```diff
@@ -0,0 +1,162 @@
+For example:
+"This cluster will be detached in 2 days."
+"The job ran for 2 hours."
+"This service will be available 01 January 2020."
```
This guide is for US English, but this is the European date format. Is that intentional? Same with showing "13:00" and not "1 pm".

mperrotti replied:
Yes, that's intentional.

mperrotti commented:
Do not merge until @danielone has approved

joinan06 commented on design-guidelines/copy-and-content/date-time-formatting.md:
```diff
@@ -0,0 +1,162 @@
+| < 1 minute | just now                   |
+| < 1 hour   | {n} minutes ago            |
+| < 24 hours | {n} hours ago              |
+| < 7 days   | {n} days ago               |
```
Hey Mike, should we consider a situation like this?
when time =24 hours, the value is '{n} day ago'
when time =1 month, the value is '{n} month ago'
...

mperrotti replied:
Sure, I will be more specific about not pluralizing when there is just 1 day/month/etc

mperrotti commented on design-guidelines/copy-and-content/date-time-formatting.md:
```diff
@@ -0,0 +1,172 @@
+
+In the rare case a time range is being used in a sentence, the en dash should not be used. For example, "The cluster was active from 12:00 until 18:00".
+
+When displaying a time range, there should be some indication for the duration of the time range. The context in which the time range is being display should inform how precise of a duration should be shown. For example, given the range "12:00:00 – 18:24:30", we could display the duration as "6 hours", "6 hours and 24 minutes", or "6 hours, 24 minutes and 30 seconds" depending on how much detail  ...
```
> let's indicate that for tables best to use abbreviated version especially when showing more than 1 level of granularity

Agreed, this is good thing to call out.

mperrotti replied:
I agree that both "6 hours, 24 minutes and 30 seconds" and "6 hours 24 minutes 30 seconds" are very long and unlikely to be used. However, I didn't want to assume we'd never want something like that, so I wanted to make sure we had that case covered.

mperrotti replied:
I had a call with Daniel where he was able to clarify what he wanted here. I'll be pushing up another change soon.

mperrotti commented on design-guidelines/copy-and-content/date-time-formatting.md:
```diff
@@ -0,0 +1,172 @@
+
+A duration should be shown when a user needs to know how much time an event took.
+
+Note: the two below tables are using abbreviated units, but the full unit name is also acceptable.
```
We cover this in the Abbreviations section

mperrotti commented:
@DanielMSchmidt - I addressed all of your feedback except for the first paragraph in [this comment](https://github.com/dcos-labs/ui-kit/pull/521#discussion_r440604215).

I don't understand what you were trying to communicate that was different from the original text. If there's something I'm missing, please explain, and I'll push another update.

danielone commented:
Yes but let’s pick one long form style for now. 

> On Jun 16, 2020, at 7:44 AM, Mike Perrotti <notifications@github.com> wrote:
> 
> ﻿
> @mperrotti commented on this pull request.
> 
> In design-guidelines/copy-and-content/date-time-formatting.md:
> 
> > +
> +### Time of day
> +
> +Times are displayed using the 24-hour clock, and a leading 0 is used for single-digit hours. For example, we would display "13:00", not "01:00 PM".
> +
> +#### Time Zones
> +
> +Time zones should not be displayed because we should opt for showing the local time. The exception to this rule is when we need to display a timestamp using the UTC time standard. When displaying a UTC timestamp, please follow the W3C formatting guidelines: [W3C Date and Time Formats](https://www.w3.org/TR/NOTE-datetime#:~:text=Times%20are%20expressed%20in%20UTC,mm%22%20minutes%20ahead%20of%20UTC.)
> +
> +### Time Ranges
> +
> +The values in a time range should be separated by an en dash with 1 space on either side. For example, "12:00 – 18:00" or "6 May – 10 May".
> +
> +In the rare case a time range is being used in a sentence, the en dash should not be used. For example, "The cluster was active from 12:00 until 18:00".
> +
> +When displaying a time range, there should be some indication for the duration of the time range. The context in which the time range is being display should inform how precise of a duration should be shown. For example, given the range "12:00:00 – 18:24:30", we could display the duration as "6 hours", "6 hours and 24 minutes", or "6 hours, 24 minutes and 30 seconds" depending on how much detail the user needs to know about the time range.
> I agree that both "6 hours, 24 minutes and 30 seconds" and "6 hours 24 minutes 30 seconds" are very long and unlikely to be used. However, I didn't want to assume we'd never want something like that, so I wanted to make sure we had that case covered.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

mperrotti commented:
Thanks @danielone - I added some clarity. Please approve if you think this is merge-able now 👍

mperrotti commented:
I think this is merge-able. Any further changes can be addressed in a follow-up.
